src/main/java/net/sf/jabref/model/entry/BibtexString.java

```diff
@@ -152,4 +152,28 @@ public String getParsedSerialization() {
     public boolean hasChanged(){
         return hasChanged;
     }
+
+    /*
+    * Returns user comments (arbitrary text before the string) if there are any. If not returns the empty string
+     */
+    public String getUserComments() {
```

Comment:
No problem, having an experienced antagonist is probably the best way to learn something :smile: 

So now ignoring everything about the actual implementation and only speaking about "business objects": There are different items which can be contained in a bib file. For example `@Comments`, `@Strings` and normal bib entries. They all have a similar structure (e.g begin with `@`, followed by some identifier and then braces which surround the actual content) and all of them could have some text comments in front of them. Thus there is also a common behavior when it comes to parsing and writing.  
So how would my deer antagonist reflect this common structure and behavior in java code?

My idea would be: BibString, BibEntry, BibComment all derive from BibItem. BibItem has methods to parse and write at least the form `@something { abstract method to parse / write value }`. Then the writer just gets a list of BibItems and invokes the write method on them. Similarly the parser returns a list of BibItems while the parsing of every item was done in the subclass. But again...this idea somehow corresponds from a `service` implementation to a full-blown domain model....so yeah, maybe we should discuss this in the next dev call ;-)